libs/pages/settings/src/lib/feature/page-organization-webhooks-feature/webhook-crud-modal-feature/webhook-crud-modal-feature.tsx

```diff
@@ -31,9 +43,15 @@ export function WebhookCrudModalFeature({ organizationId, closeModal, webhook }:
   methods.watch(() => enableAlertClickOutside(methods.formState.isDirty))
 
   const onSubmit = methods.handleSubmit(async (data) => {
+    let secretToSend = data.target_secret
+    if (hasExistingSecret && !secretTouched) {
+      secretToSend = SECRET_VALUE_UNCHANGED
+    }
```

Comment:
Instead of doing this, react-hook-form exposes a `isDirty` boolean (coming from from `fieldState`). That'd be better to use that.